src/sortStudents.ts

```diff
@@ -1,16 +1,76 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function averageGrades(grades: number[]): number {
+  return grades.reduce((prev, grade) => prev + grade, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudent: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return order === 'asc'
+        ? sortedStudent.sort((
+          student,
+          nextStudent,
+        ) => (
+          student[sortBy].localeCompare(nextStudent[sortBy])))
```

Comment:
Code would be easer to read if you rewrite kind of this 👇 
```suggestion
        ? sortedStudent.sort((student, nextStudent) => (
          student[sortBy].localeCompare(nextStudent[sortBy])))
```